jobbergate-agent/tests/jobbergate/test_compute.py

```diff
@@ -0,0 +1,307 @@
+                                aggregated_data[key][measurement] = value
+                    current_time += 10
+
+        # Create aggregated list
+        aggregated_list = cast(
+            list[tuple[int, str, str, str, float, float, float, float, float, float, float, float, float, float]],
+            [
+                (
+                    time,
+                    host,
+                    step,
+                    task,
+                    *(aggregated_data[(time, host, step, task)][measurement] for measurement in measurement_names),
+                )
+                for (time, host, step, task) in aggregated_data
+            ],
+        )
+
+        return measures, aggregated_list
+
+    return _generate_and_aggregate
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "num_points_per_measurement, num_hosts, num_jobs, num_steps, num_tasks",
+    [
+        (1, 1, 1, 1, 1),
+        (3, 10, 1, 5, 10),
+        (7, 3, 1, 4, 2),
+    ],
+)
+async def test_aggregate_influx_measures__success(
+    num_points_per_measurement: int,
+    num_hosts: int,
+    num_jobs: int,
+    num_steps: int,
+    num_tasks: int,
+    generate_and_aggregate_job_metrics_data: Callable[
+        [int, int, int, int, int],
+        tuple[
+            list[InfluxDBMeasure],
+            list[tuple[int, str, str, str, float, float, float, float, float, float, float, float, float, float]],
+        ],
+    ],
+):
+    """
+    Test that the ``aggregate_influx_measures()`` function can successfully aggregate
+    a list of InfluxDBMeasure data points.
+    """
+    measures, expected_aggregated_data = generate_and_aggregate_job_metrics_data(
+        num_points_per_measurement, num_hosts, num_jobs, num_steps, num_tasks
+    )
+
+    start_time = time.monotonic()
+    aggregated_data = aggregate_influx_measures(iter(measures))
+    end_time = time.monotonic()
+
+    print(f"Aggregated {len(measures)} data points in {end_time - start_time:.5f} seconds")
+
+    for data_point in aggregated_data:
+        assert data_point in expected_aggregated_data
+
+
+@pytest.mark.asyncio
+async def test_aggregate_influx_measures__empty_data_points():
+    """
+    Test that the ``aggregate_influx_measures()`` function returns an empty list
+    when given an empty iterator of data points.
+    """
+    data_points = []
+
+    result = aggregate_influx_measures(iter(data_points))
+
+    assert result == []
+
+
+@pytest.mark.parametrize(
+    "current, peak",
+    [(0, 0), (100, 200), (87, 100), (34, 43), (0, 98654), (3245879, 0)],
+)
+@mock.patch("jobbergate_agent.utils.compute.logger")
+@mock.patch("jobbergate_agent.utils.compute.tracemalloc")
+def test_measure_memory_usage_decorator(
+    mocked_tracemalloc: mock.MagicMock, mocked_logger: mock.MagicMock, current: int, peak: int
+):
+    """Test the measure_memory_usage decorator."""
+
+    mocked_tracemalloc.get_traced_memory.return_value = (current, peak)
+
+    @measure_memory_usage
```

Comment:
why measure the memory usage here?

Reply:
this is for benchmark purposes while developing

Reply:
Oh, I see. You are actually testing this decorator here. Disregard!